app/uk/gov/hmrc/helptosavestub/controllers/NSIGetAccountBehaviour.scala

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavestub.controllers
+
+import java.time.LocalDate
+import java.util.UUID
+
+import cats.instances.string._
+import cats.syntax.eq._
+import play.api.libs.json.{Format, Json}
+import ai.x.play.json.Jsonx
+import uk.gov.hmrc.helptosavestub.models.NSIErrorResponse
+
+object NSIGetAccountBehaviour {
+
+  def getAccountByNino(nino: String, correlationId: Option[String]): Either[NSIErrorResponse, NSIGetAccountByNinoResponse] =
+    if (nino === "EM000001A") {
+      Right(NSIGetAccountByNinoResponse.bethNSIResponse(correlationId))
+    } else if (nino === "EM000002A") {
+      Right(NSIGetAccountByNinoResponse.peteNSIResponse(correlationId))
+    } else if (nino === "EM000003A") {
+      Right(NSIGetAccountByNinoResponse.lauraNSIResponse(correlationId))
+    } else if (nino === "EM000004A") {
+      Right(NSIGetAccountByNinoResponse.tonyNSIResponse(correlationId))
+    } else if (nino === "EM000005A") {
+      Right(NSIGetAccountByNinoResponse.monikaNSIResponse(correlationId))
+    } else if (nino === "EM000006A") {
+      Right(NSIGetAccountByNinoResponse.happyNSIResponse(correlationId))
+    } else if (nino === "EM000007A") {
+      Right(NSIGetAccountByNinoResponse.takenNSIResponse(correlationId))
+    } else if (nino === "EM000008A") {
+      Right(NSIGetAccountByNinoResponse.spencerNSIResponse(correlationId))
+    } else if (nino === "EM000009A") {
+      Right(NSIGetAccountByNinoResponse.alexNSIResponse(correlationId))
+    } else if (nino === "TM739915A") {
+      Right(NSIGetAccountByNinoResponse.annaNSIResponse(correlationId))
+    } else {
+      Left(NSIErrorResponse.unknownNinoResponse(correlationId))
+    }
+
+  case class NSIGetAccountByNinoResponse(version:                   String,
+                                         correlationId:             Option[String],
+                                         accountNumber:             Long,
+                                         availableWithdrawal:       String,
+                                         accountBalance:            String,
+                                         accountClosedFlag:         Boolean,
```

Comment:
This should be a String which is `" "` for open accounts and `"C" for closed accounts according to the ICD.

Reply:
updated